### PR TITLE
Add s3 access point support (#442)

### LINKED ISSUE #442: S3 buckets are n't getting cleaned up and access points are n't deleted while nuking s3 bucket
While the S3 bucket is nuked it gave error that s3 buckets are n't getting cleaned up and access points are n't getting deleted
I am using cloud nuke v0.29.0

### PR DESCRIPTION
**Pull request by Jennas-Lee**
## Description

Fixes #442.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.
- [ ] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added / Removed / Updated [X].

- Added s3 access point support.
- Added s3 object lambda access point support.
- Added s3 multi region access point support.

### CODE REVIEW

**Comment by zackproser** [comment]
@Jennas-Lee There is one missing change in `config/config_test.go` that is causing the build to fail. Please make [this same change](https://github.com/gruntwork-io/cloud-nuke/commit/82c45151f1e19215a0c493d01257998d00311c17) on your branch to get this test passing.

File: aws/s3_access_point.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+		}
+	}
+
+	logging.Logger.Debugf("[OK] %d S3 Object Lambda Access Points deleted in %s", len(deletedNames), *session.Config.Region)
+
+	return nil
+}
+
+func getAllS3MultiRegionAccessPoints(session *session.Session, configObj config.Config) ([]*string, error) {
+	// NOTE: this action will always be routed to the US West (Oregon) Region.
```
Comment (zackproser) [inline aws/s3_access_point.go:230]: NIT: oh interesting, how come?

File: aws/s3_access_point.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+}
+
+// deleteS3MultiRegionAccessPointAsync deletes the provided S3 Multi Region Access Point asynchronously ina goroutine,
+// using wait groups for concurrency control and a return channel for errors.
+func deleteS3MultiRegionAccessPointAsync(wg *sync.WaitGroup, errChan chan error, svc *s3control.S3Control, s3Mrap *string, accountId string) {
+	defer wg.Done()
+
+	input := &s3control.DeleteMultiRegionAccessPointInput{
+		AccountId: aws.String(accountId),
+		Details: &s3control.DeleteMultiRegionAccessPointInput_{
```
Comment (zackproser) [inline aws/s3_access_point.go:377]: Is there an extra `_` here at the end of `PointInput`? :thinking: `Details: &s3control.DeleteMultiRegionAccessPointInput_{`

Comment (Jennas-Lee) [reply aws/s3_access_point.go:377]: Yes. aws-sdk-go (v1) docs says `A container for the information associated with a DeleteMultiRegionAccessPoint request`.

[https://docs.aws.amazon.com/sdk-for-go/api/service/s3control/#DeleteMultiRegionAccessPointInput_](https://docs.aws.amazon.com/sdk-for-go/api/service/s3control/#DeleteMultiRegionAccessPointInput_)

Comment (zackproser) [reply aws/s3_access_point.go:377]: Ah great, TIL - thank you!

**Comment by zackproser** [comment]
@Jennas-Lee Thanks for making the changes! This LGTM - we'll likely merge and release this shortly.

**Comment by ellisonc** [comment]
@Jennas-Lee Sorry about the delay, could you fix the merge conflicts. if you @ me once that's done I can get this moved forward quickly

**Comment by hongil0316** [comment]
Closing this PR as there has been no response for the past month or so. Feel free to re-open the PR.
It may involve quite a bit of conflict resolution required due to some refactoring changes we've made recently.

**Comment by FelipeRuizGarcia** [comment]
@zackproser  @ellisonc @Jennas-Lee seems that this requires an extra effort to rebase upstream, is there a chance u could do it ?

**Pull request closed without merging**